lib/iex/lib/iex/evaluator.ex

```diff
@@ -83,20 +83,38 @@ defmodule IEx.Evaluator do
     file = Keyword.get(opts, :file, "nofile")
     line = Keyword.get(opts, :line, 1)
     column = Keyword.get(opts, :column, 1)
+    last_op = Keyword.get(opts, :last_op, :none)
     charlist = String.to_charlist(input)
 
     result =
-      with {:ok, tokens} <- :elixir.string_to_tokens(charlist, line, column, file, opts) do
-        :elixir.tokens_to_quoted(adjust_operator(tokens, line, column, file, opts), file, opts)
+      with {:ok, tokens} <- :elixir.string_to_tokens(charlist, line, column, file, opts),
+           {:ok, adjusted_tokens} <- adjust_operator(tokens, line, column, file, opts, last_op),
+           {:ok, forms} <- :elixir.tokens_to_quoted(adjusted_tokens, file, opts) do
+        last_op =
+          Enum.reduce_while(tokens, :other, fn
+            {:match_op, _, _}, _ -> {:halt, :match}
+            _, acc -> {:cont, acc}
+          end)
```

Comment:
I would look at the forms instead. If `match?({:=, _, [_, _], forms)`, then you have a match. :)